### PR TITLE
Bug fix: avoid using caching in SurfaceSpatialIndex as this results i…

### PR DESCRIPTION
**Pull request by Ghazi-Bouabene**
…n different values for the same point when ran in parallel. Fixes issue #367

### CODE REVIEW

File: src/main/scala/scalismo/mesh/boundingSpheres/SurfaceSpatialIndex.scala [hunk 1]
```diff
@@ -188,7 +188,7 @@ private[mesh] class TriangleMesh3DSpatialIndex(private val bs: BoundingSphere,
     val p = point.toVector
 
     // last triangle might be a good candidate
```
Comment (marcelluethi) [inline src/main/scala/scalismo/mesh/boundingSpheres/SurfaceSpatialIndex.scala:190]: I guess this comment does not fit the code anymore and should be updated.

**Review by marcelluethi — COMMENTED** [review]
Thanks for the PR. I cannot assess the implications of this change and need to defer to @Andreas-Forster. If it is fine for Andreas, it is okay for me.

**Review by Andreas-Forster — APPROVED** [review]
I agree with the PR that it fixes the issue we have right now. While it might be that it gets slower, it is definitively more deterministic.

When we need the last bit of performance we would have to check if there is a way to fix this issue, by improving the algorithm. But this does not have priority.

**Comment by marcelluethi** [comment]
@Andreas-Forster I realized only now that I don't understand the issue very well. Could you quickly comment on why taking the last points as a starting point makes it non-deterministic, while taking the head does not? Is this also due to numerical instabilities or is there a bug hidden somewhere that we should fix?

**Comment by Andreas-Forster** [comment]
I did not debug it in detail. But I guess the tree traversal could be different for a point when you have a different initialization there. Just the order of visiting the nodes. I guess that sometimes then are two points found that have the (nearly) same distance and hence the order matters. Then starting always at the same point, makes the tree traversal agnostic of the last found point.

**Pull request merged**